src/managed/Microsoft.DotNet.Build.Bundle/Program.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.Build.Bundle
+{
+    /// <summary>
+    ///  The main driver for Bundle and Extract operations.
+    /// </summary>
+    
+    public static class Program
+    {
+        enum RunMode
+        {
+            Help,
+            Bundle,
+            Extract
+        };
+
+        static RunMode Mode = RunMode.Bundle;
+
+        // Common Options:
+        static bool Verbose = false;
+        static string OutputDir;
+
+        // Bundle options:
+        static bool EmbedPDBs = false;
+        static string HostName;
+        static string ContentDir;
+
+        // Extract options:
+        static string BundleToExtract;
+
+        // Typical usages are:
+        // Bundle: bundle -d <publish-dir> -a <host-exe>
+        // Extract: bundle -e <single-exe>
+        static void Usage()
+        {
+            Console.WriteLine($".NET Core Bundler ({Bundler.Version})");
+            Console.WriteLine("bundle [<mode>] [<options>]");
+            Console.WriteLine("where <Mode> is one of:");
+            Console.WriteLine("  Embed mode (by default)");
+            Console.WriteLine("  Extract mode (triggered by -e)");
+            Console.WriteLine("Embed mode options:");
+            Console.WriteLine("  -d <path>  Directory containing the files to bundle");
+            Console.WriteLine("  -a <name>  Application host (within the content directory)");
+            Console.WriteLine(" [-pdb+]     Embed the PDB file");
+            Console.WriteLine("Extract mode options:");
+            Console.WriteLine("  -e <path>  Path to the bundle file to extract");
+            Console.WriteLine("Common options:");
+            Console.WriteLine(" [-o <path>] Output directory (default: current)");
+            Console.WriteLine(" [-v]        Generate verbose output");
+            Console.WriteLine(" [-?]        Display usage information");
+        }
+
+        public static void Log(string fmt, params object[] args)
+        {
+            if (Verbose)
+            {
+                Console.WriteLine("LOG: " + fmt, args);
+            }
+        }
+
+        static void Fail(string type, string message)
+        {
+            Console.Error.WriteLine($"{type}: {message}");
+        }
+
+        static void ParseArgs(string[] args)
```

Comment:
Please get @KathleenDollard and @richlander to review the command line options - it's a public CLI/UI after all.